### DISCUSSION TITLE
Feature Request: `RENAME`-support with a star-expression, similar to `EXCLUDE`/`REPLACE`

### DISCUSSION THREAD

**Discussion by taffit** [duckdb/duckdb#14376] [Ideas]
Having a table with a lot of columns, the star expressions saves a lot of time and writing when selecting the needed columns. When you need to do calculations to the selected columns AND need to rename just some single columns, this is not possible without explicitely listing all the columns once. The `REPLACE`-statement allows just replacing with the same name (else it complains). Using a regex with the `COLUMNS()` expression allows renaming using capture groups, however this works only, if all the columns follow the same schema.        

For example, I want to rename the single column `sold_today` to `sold`, while still doing the calculation with the star or `COLUMNS()`-expression on all the selected columns:
```sql
SELECT sum(COLUMNS('(prefix_)?(some_col\d{1,3}|sold_today)(_usd)'))
                              EXCLUDE(comment_usd)  -- works
                              RENAME(sold_today as sold)  -- does not work
FROM mytable
GROUP BY ALL
```

I just found some references in the issue where [`EXCLUDE`/`REPLACE` was discussed](https://github.com/duckdb/duckdb/issues/2199).    
And just for information purpose, [Snowflake has it implemented](https://docs.snowflake.com/en/sql-reference/sql/select) in a similar way to `EXCLUDE`/`RENAME`.

And as I can't thank you enough, here just once more: Thank you! Thank you! Thank you, for the gem that duckdb is. It made so many things easier for me, and made task possible that I couldn't realize before easily ... Thank you!

**Comment by Tishj** [comment]
I made a quick start on this:
<https://github.com/Tishj/duckdb/tree/star_expression_rename>

from the branch:
```sql
# name: test/sql/parser/test_star_rename.test
# group: [parser]

statement ok
create table tbl (
	prefix_some_col123 INTEGER,
	comment_usd VARCHAR,
	sold_today BOOL
);

statement ok
insert into tbl values(1, 'test', true)

query II
SELECT alias(COLUMNS(*)) FROM (
	SELECT * EXCLUDE(comment_usd) RENAME (sold_today as sold)
	FROM tbl
)
----
prefix_some_col123	sold
```

**Comment by Mytherin** [comment]
This is implemented now in https://github.com/duckdb/duckdb/pull/14650
